File: NefyraExo/app/src/main/java/com/nefyra/exo/JsonUtil.java
package com.nefyra.exo;

import android.content.Context;
import android.util.Log;
import java.io.File;
import java.io.FileInputStream;
import java.io.InputStream;
import com.fasterxml.jackson.databind.ObjectMapper;
import com.nefyra.exo.VideoConfig;

public class JsonUtil {
  public static VideoConfig readUrlFromPrivateStorage(Context context) {
    FileInputStream fis = null;
    try {
      File file = new File(context.getFilesDir(), "video_config.json");

      if (!file.exists()) {
        Log.e("JSON", "File not found in private storage");
        return null;
      }

      fis = new FileInputStream(file);
      return new ObjectMapper().readValue(fis, VideoConfig.class);
    } catch (Exception e) {
      Log.e("JSON", "解析失败", e);
      return null;
    } finally {
      try {
        if (fis != null) fis.close();
      } catch (Exception e) {
        Log.e("JSON", "关闭流失败", e);
      }
    }
  }
}

File: NefyraExo/app/src/main/java/n/nefyra/exo/JsonUtil.kt
package com.nefyra.exo

import android.content.Context
import android.util.Log
import com.fasterxml.jackson.databind.DeserializationFeature
import com.fasterxml.jackson.databind.MapperFeature
import com.fasterxml.jackson.databind.ObjectMapper
import java.io.File
import java.io.FileInputStream

object JsonUtil {
    fun readUrlFromPrivateStorage(context: Context): VideoConfig? {
        var fis: FileInputStream? = null
        return try {
            val file = File(context.getExternalFilesDir(null), "video_config.json")
            val mapper = ObjectMapper()
            mapper.configure(DeserializationFeature.FAIL_ON_UNKNOWN_PROPERTIES, false)
            mapper.configure(MapperFeature.ACCEPT_CASE_INSENSITIVE_PROPERTIES, true)

            if (!file.exists()) {
                Log.e("JSON", "File not found in external private storage")
                return null
            }

            fis = FileInputStream(file)
            mapper.readValue(fis, VideoConfig::class.java)
        } catch (e: Exception) {
            Log.e("JSON", "解析失败", e)
            null
        } finally {
            try {
                fis?.close()
            } catch (e: Exception) {
                Log.e("JSON", "关闭流失败", e)
            }
        }
    }
}